frontend/src/libs/components/audio-player/audio-player.tsx

```diff
@@ -1,61 +1,117 @@
-import { useCallback, useRef, useState } from '#libs/hooks/hooks.js';
+import {
+  useAppForm,
+  useCallback,
+  useEffect,
+  useRef,
+} from '#libs/hooks/hooks.js';
 import {
   TRACK_INCREMENT_INDEX,
   TRACK_START_TIME,
 } from '#pages/meditation/libs/constants/constants.js';
 
 import { AudioControls } from './components/audio-controls/audio-controls.js';
 import { ProgressBar } from './components/progress-bar/progress-bar.js';
+import { DEFAULT_AUDIO_OPTIONS } from './libs/constants/constants.js';
+import { AudioOptionKey } from './libs/enums/enums.js';
+import { type ReferenceProperties } from './libs/types/types.js';
 
 type Properties<T> = {
-  src: string;
+  sourceName: keyof T;
+  durationTimer: number | null;
   trackIndex: number;
   onSetTrackIndex: (index: number) => void;
-  onSetCurrentTrack: (track: T) => void;
   tracks: T[];
 };
 
 const AudioPlayer = <T,>({
-  src,
+  sourceName,
```

Comment:
let's remove generic here and define basic type needed for track - like
{ name, mediaUrl }
```suggestion
const AudioPlayer = ({
```